Tests/BotTests/MergeService/DispatchServiceTests.swift

```diff
@@ -140,6 +140,88 @@ class DispatchServiceTests: XCTestCase {
+                .mergeIntoBranch { head, base in
+                    expect(head.ref) == MergeServiceFixture.defaultBranch
+                    expect(base.ref) == developBranch
+                    return .success
+                },
+
+                .postComment(checkComment(2, "Your pull request was accepted and it's currently `#1` in the `develop` queue, hold tight ⏳")),
+
+                // Note that here we shouldn't have any API call for PR#3 since it doesn't have the integration label
+                
+                .getPullRequest(checkReturnPR(dev1.with(mergeState: .clean))),
+                .getCommitStatus { pullRequest in
+                    expect(pullRequest.number) == 1
+                    return CommitState.stub(states: [.success])
+                },
+
+                .mergePullRequest(checkPRNumber(1)),
+                .deleteBranch(checkBranch(dev1.reference.source)),
+                .getPullRequest(checkReturnPR(dev2)),
+                .mergePullRequest(checkPRNumber(2)),
+                .deleteBranch(checkBranch(dev2.reference.source)),
+            ],
+            when: { service, scheduler in
+
+                scheduler.advance()
+
+                service.sendPullRequestEvent(action: .synchronize, pullRequestMetadata: dev1.with(mergeState: .blocked))
+
+                scheduler.advance()
+
+                service.sendPullRequestEvent(action: .labeled, pullRequestMetadata: dev2)
+
+                scheduler.advance()
+
+                service.sendPullRequestEvent(action: .opened, pullRequestMetadata: rel3)
+
+                scheduler.advance()
+
+                service.sendStatusEvent(state: .success)
+
+                scheduler.advance(by: .seconds(60))
+            },
+            assert: {
+                expect($0) == [
+                    .created(branch: developBranch),
+                    .state(.stub(targetBranch: developBranch, status: .starting)),
+                    .state(.stub(targetBranch: developBranch, status: .ready, pullRequests: [dev1.reference])),
+                    .state(.stub(targetBranch: developBranch, status: .integrating(dev1))),
+                    .state(.stub(targetBranch: developBranch, status: .runningStatusChecks(dev1.with(mergeState: .blocked)))),
+                    .state(.stub(targetBranch: developBranch, status: .runningStatusChecks(dev1.with(mergeState: .blocked)), pullRequests: [dev2.reference])),
+
+                    // MergeService is always created when we receive a new PR event…
+                    .created(branch: releaseBranch),
+                    .state(.stub(targetBranch: releaseBranch, status: .starting)),
+                    // … but since that new PR got filtered out for not having the integration label…
+                    .state(.stub(targetBranch: releaseBranch, status: .idle)),
+                    // … then the service is destroyed right away.
+                    .destroyed(branch: releaseBranch),
```

Comment:
Notice the new behavior: with the new implementation that avoids `DispatchService` having to know about the internal logic and impl details of `MergeService`, that means that it always creates a `MergeService` even if that `MergeService` ends up deciding there's nothing to do and gets immediately destroyed 🤷‍♂

Reply:
For now... we shouldn't exclude doing something in the future, imagine if we introduce the ability to monitor the PRs to check the number of reviews pending and notify that the PR is ready to be integrated? e.g. after 2 approvals In that scenario the service may stay alive.